### PR TITLE
Introduce JERRY_UNUSED() instead of __attr_unused___

### PR DESCRIPTION
**Pull request by bzsolt**
JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/jrt/jrt.h [hunk 1]
```diff
@@ -81,6 +80,8 @@ extern void __noreturn jerry_unimplemented (const char *, const char *, const ch
 #define JERRY_ASSERT(x) do { if (false) { (void)(x); } } while (0)
 #endif /* !JERRY_NDEBUG */
 
+#define JERRY_UNUSED(x) (void)(x)
```
Comment (LaszloLango) [inline jerry-core/jrt/jrt.h]: `#define JERRY_UNUSED(x) ((void) (x))`

File: jerry-core/jrt/jrt.h [hunk 2]
```diff
@@ -61,7 +60,7 @@
 #define JERRY_STATIC_ASSERT_GLUE(a, b, c) JERRY_STATIC_ASSERT_GLUE_ (a, b, c)
 #define JERRY_STATIC_ASSERT(x, msg) \
   typedef char JERRY_STATIC_ASSERT_GLUE (static_assertion_failed_, __LINE__, msg) \
-  [ (x) ? 1 : -1 ] __attr_unused___
+  [ (x) ? 1 : -1 ] __attribute__((unused))
```
Comment (LaszloLango) [inline jerry-core/jrt/jrt.h]: Why cannot use JERRY_UNUSED here?

Comment (bzsolt) [inline jerry-core/jrt/jrt.h]: I think it looks weird, so I kept the attribute here.
Do you have a sophisticated solution? :)

File: tests/unit/test-date-helpers.c [hunk 3]
```diff
@@ -32,9 +32,10 @@
  * Unit test's main function.
  */
 int
-main (int __attr_unused___ argc,
-      char __attr_unused___ **argv)
+main (int argc, char **argv)
```
Comment (zherczeg) [inline tests/unit/test-date-helpers.c]: I think you can simply delete both arguments, and just use:

    main ()
    {
    }

Several similar cases below as well.

**Comment by zherczeg** [comment]
I like this patch and I think it is close to ready

**Comment by LaszloLango** [comment]
LGTM

File: jerry-core/jrt/jrt.h [hunk 4]
```diff
@@ -60,8 +59,7 @@
 #define JERRY_STATIC_ASSERT_GLUE_(a, b, c) a ## b ## c
```
Comment (zherczeg) [inline jerry-core/jrt/jrt.h]: Wouldn't it be better to add an extra underscore (_) ?

    #define JERRY_STATIC_ASSERT_GLUE_(a, b, c) a ## b ## _ ## c

Comment (bzsolt) [inline jerry-core/jrt/jrt.h]: That's a good idea, it would be more readable.

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**